independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -609,29 +618,40 @@ protected static String toClassName(Type indexType, ClassInfo currentClass, Clas
                 if (typeVariable.bounds().isEmpty()) {
                     return Object.class.getName();
                 }
-                int pos = -1;
-                for (int i = 0; i < currentClass.typeParameters().size(); ++i) {
-                    if (currentClass.typeParameters().get(i).identifier().equals(typeVariable.identifier())) {
-                        pos = i;
-                        break;
-                    }
-                }
-                if (pos != -1) {
-                    List<Type> params = JandexUtil.resolveTypeParameters(actualEndpointClass.name(), currentClass.name(),
-                            indexView);
 
-                    Type resolved = params.get(pos);
-                    if (resolved.kind() != Type.Kind.TYPE_VARIABLE
-                            || !resolved.asTypeVariable().identifier().equals(typeVariable.identifier())) {
-                        return toClassName(resolved, currentClass, actualEndpointClass, indexView);
-                    }
-                }
-                return toClassName(typeVariable.bounds().get(0), currentClass, actualEndpointClass, indexView);
+                return toClassName(resolveTypeVariable(typeVariable, currentClass, actualEndpointClass, indexView),
+                        currentClass, actualEndpointClass, indexView);
             default:
                 throw new RuntimeException("Unknown parameter type " + indexType);
         }
     }
 
+    private static Type resolveTypeVariable(TypeVariable typeVariable, ClassInfo currentClass, ClassInfo actualEndpointClass,
+            IndexView indexView) {
+        if (typeVariable.bounds().isEmpty()) {
+            return Type.create(DotName.createSimple(Object.class.getName()), Kind.CLASS);
+        }
+        int pos = -1;
+        for (int i = 0; i < currentClass.typeParameters().size(); ++i) {
```

Comment:
Perhaps `currentClass.typeParameters()` can be extracted to a local variable here

Reply:
Done